include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -454,6 +454,21 @@ def NamedConstantOp : MooreOp<"named_constant", [
   }];
 }
 
+def StringConstantOp : MooreOp<"string", [Pure, ConstantLike]> {
+  let summary = "Produce a constant string value";
+  let description = [{
+    Produces a constant value of string type.
+
+    Example:
+    ```mlir
+    %0 = moore.string "hello world"
+    ```
+  }];
+  let arguments = (ins StrAttr:$value);
+  let results = (outs SimpleBitVectorType:$result);
+  let assemblyFormat = "$value attr-dict `:` type($result)";
```

Comment:
You may not need type($result) once result is guaranteed to be StringType always.

Reply:
Could we make this return an `IntType` instead? I'm pretty sure the string literal will always have a type `bit [N-1:0]`, which would be a `!moore.iN` integer type.